device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -172,14 +175,28 @@ public void registerMultiplexedDevice(DeviceClientConfig config)
         deviceClientConfigs.add(config);
     }
 
-    public void unregisterMultiplexedDevice(DeviceClientConfig config)
+    /**
+     * Asynchronously unregister a multiplexed device from an active multiplexed connection or synchronously unregister.
+     * a multiplexed device from a closed multiplexed connection.
+     * @param config the config of the device that should be unregistered.
+     * @param willReconnect true if the device will be re-registered soon because it is reconnecting.
+     */
+    public void unregisterMultiplexedDevice(DeviceClientConfig config, boolean willReconnect)
```

Comment:
Adding a flag to the unregister call that signals to this AMQP layer to cache the unregistered session objects since we will re-register that device session soon.